lib/logstash/filters/truncate.rb

```diff
@@ -40,13 +40,17 @@ def register
 
   def filter(event)
     if @fields
-      @fields.each do |field|
-        Truncator.truncate(event, field, @length_bytes)
-      end
+      truncated = Truncator.truncate_all(fields, event, @length_bytes)
     else
-      Truncator.truncate_all(event, @length_bytes)
+      # TODO(sissel): I couldn't find a better way to get the top level keys for
+      # an event. Nor could I find a way to iterate over all the keys in an
+      # event, so this may have to suffice.
+      fields = event.to_hash.keys.map { |k| "[#{k}]" }
+      truncated = Truncator.truncate_all(fields, event, @length_bytes)
     end
-    filter_matched(event)
+    
+    @logger.debug("truncated one or more fields from event to length #{@length_bytes}") if truncated
+    filter_matched(event) if truncated
```

Comment:
Instead of double `if`s is better to collapse in only one

```suggestion
    if truncated
      @logger.debug("truncated one or more fields from event to length #{@length_bytes}")
      filter_matched(event)
    end  
```